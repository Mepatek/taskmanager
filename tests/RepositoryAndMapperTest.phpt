<?php
/**
 * TEST: Test all providers with all mappers
 */

namespace Test;

use Mepatek\TaskManager\Entity\TaskAction_ITask;
use Mepatek\TaskManager\Entity\TaskCondition_Cron;
use Mepatek\TaskManager\Entity\TaskHistory;
use Nette,
	Tester,
	Tester\Assert;

require __DIR__ . '/bootstrap.php';


class RepositoryAndMapperTest extends Tester\TestCase
{
	const TRANSACTION = true;

	private $database;

	private $varchar20;
	private $varchar30;
	private $varchar40;
	private $varchar50;
	private $varchar60;
	private $varchar100;
	private $varchar150;
	private $varchar200;
	private $varchar250;
	private $varchar255;
	private $binaryData;
	private $text;
	private $datetime;
	private $float;

	function __construct()
	{
	}


	function setUp()
	{
		$connection = new \Nette\Database\Connection("sqlite:" . __DIR__ . "/data/TaskManager.db",
			null, null);
		$structure = new \Nette\Database\Structure( $connection, new \Nette\Caching\Storages\FileStorage(TEMP_DIR) );
		$conventions = new \Nette\Database\Conventions\DiscoveredConventions( $structure );
		$this->database = new \Nette\Database\Context($connection, $structure, $conventions);


		$this->varchar20 = (string)Nette\Utils\Random::generate(20);
		$this->varchar30 = (string)Nette\Utils\Random::generate(30);
		$this->varchar40 = (string)Nette\Utils\Random::generate(40);
		$this->varchar50 = (string)Nette\Utils\Random::generate(50);
		$this->varchar60 = (string)Nette\Utils\Random::generate(60);
		$this->varchar100 = (string)Nette\Utils\Random::generate(100);
		$this->varchar150 = (string)Nette\Utils\Random::generate(150);
		$this->varchar200 = (string)Nette\Utils\Random::generate(200);
		$this->varchar250 = (string)Nette\Utils\Random::generate(250);
		$this->varchar255 = (string)Nette\Utils\Random::generate(255);
		$this->text = (string)Nette\Utils\Random::generate(8000);
		$this->binaryData = (string)Nette\Utils\Random::generate(2048);
		$this->datetime = new \Nette\Utils\DateTime();
		$this->datetime1 = new \Nette\Utils\DateTime("2015-12-12");
		$this->date = new \Nette\Utils\DateTime("2015-12-10");
		$this->float = ((float)Nette\Utils\Random::generate(10, "0-9")) / 1000;
	}


	function testTaskRepositoryNetteDatabaseMapper()
	{
		$mapper = new \Mepatek\TaskManager\Mapper\TaskNetteDatabaseMapper($this->database);
		$repository = new \Mepatek\TaskManager\Repository\TaskRepository($mapper);

		$this->beginTransaction();

		$task = new \Mepatek\TaskManager\Entity\Task;

		$task->name = $this->varchar150;
		$task->created = $this->datetime;
		$task->source = $this->varchar255;
		$task->author= $this->varchar100;
		$task->description = $this->text;
		$task->deleteAfterRun = TRUE;
		$task->maxExecutionTimeInSecond = 88;
		$task->state = 0;
		$task->disabled = FALSE;
		$task->nextRun = $this->datetime;
		$task->lastRun = $this->datetime;
		$task->lastSuccess = TRUE;

		$action = new TaskAction_ITask;
		$action->order = 1;
		$action->className = $this->varchar50;
		$action->nameSpace =  $this->varchar100;
		$action->arguments = array();
		$task->addAction( $action );

		$condition = new TaskCondition_Cron;
		$condition->created = $this->datetime;
		$condition->expired = $this->datetime;
		$condition->active =  TRUE;
		$condition->cronExpression = $this->varchar50;
		$task->addCondition( $condition );

		Assert::true( $repository->save($task) );

		$itemId = $task->id;
		Assert::type("integer", $itemId);

		$item1 = $repository->find($itemId);
		Assert::type("Mepatek\\TaskManager\\Entity\\Task", $item1);
		Assert::equal($task, $item1);

		$repository->setStateRunning($task);
		$item1 = $repository->find($itemId);
		Assert::type("Mepatek\\TaskManager\\Entity\\Task", $item1);
		Assert::type("Nette\\Utils\\DateTime", $item1->exceedDateTime);
		Assert::equal($task, $item1);

		// update
		$item1->name = "1";
		$item1->source = "1";
		$item1->author= "1";
		$item1->description = "1";
		$item1->deleteAfterRun = FALSE;
		$item1->state = 0;
		$item1->nextRun = $this->datetime1;
		$item1->lastRun = $this->datetime1;
		$item1->lastSuccess = FALSE;
		Assert::true($repository->save($item1));

		// find
		$item2 = $repository->find($itemId);
		Assert::equal($item1, $item2);
		Assert::notEqual($task, $item2);

		// delete
		Assert::true($repository->deletePermanently($itemId));

		// find - NOT FOUND (NULL) is OK
		Assert::null($repository->find($itemId));

		$this->rollBack();
	}


	function testTaskHistoryRepositoryNetteDatabaseMapper()
	{
		$mapper = new \Mepatek\TaskManager\Mapper\TaskNetteDatabaseMapper($this->database);
		$repository = new \Mepatek\TaskManager\Repository\TaskRepository($mapper);

		$mapperHistory = new \Mepatek\TaskManager\Mapper\TaskHistoryNetteDatabaseMapper($this->database);
		$repositoryHistory = new \Mepatek\TaskManager\Repository\TaskHistoryRepository($mapperHistory);

		$this->beginTransaction();

		$task = new \Mepatek\TaskManager\Entity\Task;

		$task->name = $this->varchar150;
		$task->created = $this->datetime;
		$task->source = $this->varchar255;
		$task->author= $this->varchar100;
		$task->description = $this->text;
		$task->deleteAfterRun = TRUE;
		$task->state = 1;
		$task->disabled = FALSE;
		$task->nextRun = $this->datetime;
		$task->lastRun = $this->datetime;
		$task->lastSuccess = TRUE;

		Assert::true( $repository->save($task) );

		$taskHistory = new TaskHistory;
		$taskHistory->taskId = $task->id;
		$taskHistory->started = $this->datetime;
		$taskHistory->finished = $this->datetime;
		$taskHistory->resultCode = 0;
		$taskHistory->user = $this->varchar150;
		$output = new \Mepatek\TaskManager\Entity\Output();
		$output->write($this->text);
		$taskHistory->output = $output;

		Assert::true( $repositoryHistory->save($taskHistory) );

		$itemId = $taskHistory->id;
		Assert::type("integer", $itemId);

		$item1 = $repositoryHistory->find($itemId);
		Assert::type("Mepatek\\TaskManager\\Entity\\TaskHistory", $item1);
		Assert::equal($taskHistory, $item1);

		// update
		$item1->started = $this->datetime1;
		$item1->finished = $this->datetime;
		$item1->resultCode = 2;
		$item1->user = "1";
		$output->output = "1";
		$item1->output = $output->output;
		Assert::true($repositoryHistory->save($item1));

		// find
		$item2 = $repositoryHistory->find($itemId);
		Assert::equal($item1, $item2);
		Assert::notEqual($task, $item2);

		$repositoryHistory->fillTask($task);
		Assert::equal($task->history[0], $item2);

		// delete
		Assert::true($repositoryHistory->delete($itemId));

		// find - NOT FOUND (NULL) is OK
		Assert::null($repositoryHistory->find($itemId));

		$this->rollBack();
	}

	/**
	 * If TRANSACTION is set to TRUE, begin database transaction
	 */
	private function beginTransaction()
	{
		if (self::TRANSACTION) {
			$this->database->beginTransaction();
		}
	}

	/**
	 * If TRANSACTION is set to TRUE, rollBack
	 */
	private function rollBack()
	{
		if (self::TRANSACTION) {
			$this->database->rollBack();
		}
	}
}


$test = new RepositoryAndMapperTest();
$test->run();
