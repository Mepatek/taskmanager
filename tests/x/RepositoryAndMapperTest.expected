Mepatek\TaskManager\Entity\TaskHistory::__set_state(/* #2209 */ array(
	'id' => 1,
	'taskId' => 34,
	'started' => Nette\Utils\DateTime::__set_state(/* #03db */ array(
		'date' => '2016-03-12 12:02:39.000000',
		'timezone_type' => 3,
		'timezone' => 'Europe/Prague',
	)),
	'finished' => Nette\Utils\DateTime::__set_state(/* #4271 */ array(
		'date' => '2016-03-12 12:02:39.000000',
		'timezone_type' => 3,
		'timezone' => 'Europe/Prague',
	)),
	'resultCode' => 0,
	'user' => 'dxvwu9pfm4ahsa3mgcw0mfl2boenjw7g0kra8hcc1e5ke2cdp0q1cj3vtk42g9cphn2g4khwcal0sy7v0nzckrg59hew3j1cpdrjvjdhs0n8d7gdctkc4a9ewib05byfmp8wl29dma1blav860cct4',
	'output' => Mepatek\TaskManager\Entity\Output::__set_state(/* #6fbf */ array(
		'output' => "2rs85r9dxjh1iaq5giqsie8qmmwkdle86i67uh6goehvu5me3h2qsri851jciprjefkmp1py59i4gwd5njcb0ee8k0sdbpq79mqjs1dpuemb719teup77ddfzeplyb5s9lry19xk71a5xo257yzkj9u9in98gal61vfce607w0803yjxp1h9vskpn4qx5jba668v9y4hmue2qdzg5lu4jnnx3wx5ww3lmj0zvx16ptrcc35e3nva4lk08cl31nbdki26vb193zgm1omxiavp7xk9zk2pcfn2leuzo8w0j7zlz05vc7zo800paow3g8a7clr3avktyjs36vb6xb4h9sqlp6qkl5nbfkqu2zneuiqgx61hake3cto9ymi20ifdomcjsx725mzmt2i024thrb0yh50eln5e715onetm842iwpmojchgnwyxjsd9ig5k7vbfjo10cw26ht0kn1opno5nhwyw9l0f8j2cdfvxb1ssog2z564uap6a50bv0c8doogkhh8b2a9clpv42v89bg9nczwxdha7m9q3bpkin6z9hwcedtujok1oqrp9c2t00c8o4iqhy04mwcx7b8qi9xd8bjupuz98fc6dupdn82kxk4rm9b3wrurwdkr5b0o7t5xdxewpa8vk605qr0hscr6wxlyhqf2pmzwr4weu5wj0q9up6tbu8rv3q33y0pquxfl2oypap7gl7qbx9nlknf3uywhajmuuofgm3l5smrk16v9ebk0165otjadarshwlj0ffn9yrga1hx2econb1bfdqibihn8bkz3m1uaapbnjgwsvjo7yd9vwywexuh18nytkr6pczi8qlfli1l6cyind7d6p58z1gg5g1zfh499yzzljsmyjuc2p3iyqbvuv75q7zq8eo5drgi96fjg597uteimjqri6j49y4yb6wzdqtfk71d0v1huzn7my4s197agw5xzg46qnrs1tyhnn59mwycverzguel1vxl98nqwyqi1hueq1twon6ddu0k2hhp46ynd8bpg8wqly2ps23d4rit8c9pczod9v4jmnypeppcogjeu102xycpgu5621m6qf7iyixa4w9c8y8ennqgtx2ash76l92gbbwagbfg94gui30a1l0jmfgpt6gxq1tp7mxbo3zcvo1xjjtaxvqx6rbssgleujey3zc1y8rhudczgwhoqg81ocjcx4xtmhkwfv0ml37j7q7je9a33dqsma6wo7jnbdgft30fjnu4p1i5q6xjgp8qp7sog5f0f2b4eorcvx0ydarurutrknzuu81exy33m4bxua3ouna81n709plyqx0ihshhsz5n7kmie3bpwlsoa4y5ylnlq8y783jj17alr2cgn8x7tg311o4rgakguvj0uumoztzrtl3ldgh7gt0tymxxpdf2qw2hxqv7ox5yhm4p6opd07h39g67kvc65m4mkirsmx78u67o0y5ht2gbtbhg0sx9pmvrai6vcqrnne2mxy8jb8zgenzvxwolwlzul4kcjlryq7kkw0ga55stdaht4xzuirkaqzcgfv6ohbkdifd6gwngh6lnpu07rzn6e1s2jjq8fa5rvg4hmk99aeg6e5a4skmdiov5qo5k8mhshisdnddmtepfnj88mlwmr1vleupvk19jy3eaz8fvzw47ksz3a2c9mdmsdlq2xquks166e1yf18a5rtqvnxgkzs9cj8u5fj5ibow4zb8bnr4dsqen8xj2s8ye3ih4nuqnr2741ua7dl3wqee0n21s8ixawm7nrlcqvvbofqh81lguhqzx8knnzlkijnlninj4zmk5e1745u138mgtax3igrtog3jeu6eja429x3x8ueu6lp4bt0oc5cf1vypbnq1ha123vaqq807c15q7roimf3j0dup3s6fu54450bq579t7ck1nowhpto50d0xak5s6v7qxmp8f70a95cu81pyuh0qmxstygkbyx1d86293bbf9iyl643f7bdbfjqnl946tkulcoozp97dkv03soj5pnac53w2r33rbm2eu5gbissf7a573yh7miso0n41ja4deid5j8r95vyoh8f0vz6ubs0vagbx5ykeo1z6zcdt39inhio9trrv6ae0vf4p5aebdxtf3fewhtoyj2jmh7szibij9l2yey9glqh9pes26jnb4g3nm2lug9re7crqswcgciy4raqskc5gbklo3hjbr5c6cvrny0u05y94sn2ugx4gyvazn9dnq60p5ivjz9fr41ekzunoirisnn41cxnn6rpg1j1cyvmegsl32bg7ehys7ovppikr4rvov6nydh287xoayeik97rlpu2eaeuncepenxcryj7y05hb59xotb8rzjqqyw99cjbl0o6bvvrpkpcta9ut8bpu848360ykyfrienufx7m2c73sf0wzodaf54wb4w3z4swn48zcz282b72mtrsk9vewj3tyu799kbyw5osqxe5rqez3z9lz0sffxj2cxl9e2d7nmwk6n7quq6wuhv271py4vg3bjettjtgv3v1yw3yjde02kc01nibq9j5wjd6zuli12xxa6y92g61rpev5th471j8g6xvuxi1jnsnsr5rkmwfkawaucflvvyj5jaw1oi9d0s68crdex8gy1deu2p9v90ogpj0gviwihhzd4lieq1dr3pfth28ccyxy1fq74ml7keial5brup80mq5pim2osbjno8omlgjdykkmecuum7qu54vp2qzxfrf51z28794thkur1ucm7jzybtuq8bbzgd8nusq00ljcnkokis1khhr1dgpjn7hqkoxv1fl9ky1d5x75drqnb8970ou8ey3c13hfe3ha0ffu8q1eqz8kc6tk83dwki4ns1gm0l8tkh9ns3ryk4xk7ipqpzdeodjmnvi5rb0c2pvi01yifuzubca6lj25c66sxz6xfrnewlrxyn4vhimm5fhc7o33onvyqn0b378umc5xgfimkma14gupa5zx5uzqzbzy9v2woba1w1q7bpz7ll8ust092sv23qcp5bxhr1hr98ye3ttwio3in4ad0o8p79uhrm3c3v20rx58gtjhag372oshcqo6xalhe8cvrougj5err3nurja8dugc3kaoigpgkyyqlsb1de1ddclpt0xcxce0itezfdoe74auy4ge8nhvmwhabbvizn69a343nqltrsrm3g8zcdv0blrxz458u5fg3v0xnl9ieqjqmicya86qqsyvymeyl1jnfobyz77dkxxsc7kuv3rvp7isxhqrnb0aia6oxauutlxu4g5j344bo2od7eu4adruz3408yt89e3cdvg1130j17kago8lkqred64zfpc7zvxy597we0ezkajliia8hfwhjipnv8wtrms3wgyyheb3lk97f6fghkjy4mx6j0tlh3u803k3ffe6ca4i4ufb9qk2zfy7x57ncr4z60pedfiu2mwx82zwnz8ndnhxfgmhbjfjd36b0ozf569fnsl3avxe4e6q4ejmbpa4cvbxwz05dmm4x32murbit81i02vjk9kuusds9hs904jtjb23q05zjbt47tdiln5x9clizvg5qb1r2hd027dhs5y0uqggti69mwgossk7cx2chkhx153d8ecgd26ymlwautmuypnlbewkc88nk1i2at1r93at9rb1b1aetkyeq4osmwvz9xez4vrspubs39h0zdfhnui1ynxsie7z7no3uadn22u57iquefl4dcfpiv7fxhkmiih23uppesh38qz2ulvfodn3j1d4toi407y4dfbaq11d6c4tlhb5phv6aai2llpq1j6carbsy17za7jjur0u7dly7m3be3dzcowzsqc02lwuv2nylxswcse9kczgxp3yplf8aq8drnaqdy13tmzfp87visejro3qcfj6waqqz3kcyntqf15n4vd3e49ggomfys9favd45q7z7zdayhrsou6dsaw65c0l2zeu7vy4dtdj2plu9oui78frl199vkcdoz2qrs8mtev6cobtsn37bffq2083nkryaplzphdya3c38gl3ww1fexopg5mx74unfr7rwqg23tmfoyl7gt6acu5kwwdtcb7hurxlqj0ytvp9tuyphtgswfcyjp5qwr1q7ggf0isik9qpaok4bhbr6zqd2ab3p2l6p0puj3x5ptc8y9wh49gm7lgzzhthtn0icel49dsjfuik0pej2k8oyfmbooyywqxhuz9u5lfvofyni1lfyqu9fglrqui7qw2sn4dzn0yi0m195a4d3c94j46sz31dcpqd5fw8aawt59de3yb0ou4nbw1qxmlzbfuai9sxjo667j2t04i3gdeiuv2z9qz0ssgwblph8fm5sirmeluc0j7jnfmnsyju6z56h7332g3ol03cupt43ttfbw7lz3lojvy617icvh7s0lp7o9j1g5i01hnwk0uxpxd93yxvudg35hijfhb1yto3v7ys8lh9xwmzo3zm9ra99lsmkgrbe4fgtxj2ztligpj6yr40k608b3pc0xhc8xqe6ex34qgulvxk9z9qc3i66p7g8ivh6gkd5cq7gmuu4atws4nfho5n9lesjucoavzf3g00cupb68lxx8svg6u9t696nu80rm878ra0rqjjahyiu5144eu09xskcqjhqw8j56j4epvrof2208bpqo87p2dm1efbvhw1lcphfkzulxuz3qhwrytlhewxlawfm0eg138z2fkved71fjnsjhtbu1cdb11ajhgcrm83mefl0cdijlo0gev0ddhfzcbtrg3m7v8a65ejmgc59fh9tjqi1blq8absxmd624h4yp2zojnjp1mk403nenrb2aj2q7up1pijo95c1avjpnuyoua6odswvxvrnqt4pz3wloyfh8tuu07h6zx1ce2ywi9vrwe9i7z1t6o2ho6gr1rjwn30hs39kb0e7c4wk7amdut0aupbegfyrtszd60qrhqtc6oqrs2qhyfau82d4kabw9uxqm19ib86s1j4346plkkb15hb1bugxlc8nvh78joh2p7r9nudhbyakeqndnwbdmyiwp35tnprw4srbgzmo5fbioxbi12zf52b907qxeond2v4j7msmhv8aonkub62lwleqp858hj4vxx8leesmxaxx6vf2kc8p148cr3ucmu1a9gsxablps7oq897q2us38jgt1c9906qg9hmyhysw8j7bpa00d9xoabygk49jsr1n7y5mruyji49e80isuwc0fphv5vtanyzu5wt8ijf412cc5l2kz2awf1ub6hiwzc4qoyzym83jcvko1y8lmkd9j83iulcp7tzg3wvfx9r95iqfccnc8pkwt3j4h78brulthqtxe9z0xa4404057lerc6hn0lwaolcvyat74bug566q8e467lm9jkhusg5fryxlxxky2m033cv5molv49m1bb5bn8pcxmilzu58gq5sf1re62h3vgho7yzk37oxi6diolr2ep8ic20nuu7fg01u4hvxnqxlo95j6k3ss174bew0iflz02vubtbty7rizbeq82qvxj59nm1kun7h82i6qd9mrq62ldfg41xa8s54lfhn6x7jge3ues5oxpn809w7e0e7wq8hvfk66n43ehlg30jkjttrpkqqs8fho2hoxej4jhr3qi8opbmnx1r4hy6svqkmwktj3yzs5k9lwmytxgyjrgvw65ly8z5hyszwkojb8n0e9dpcxaq1scajkx0ixalfqyoirlqyyeoznoi5r963s452vikix62quhdifpjz1c1jp79ucqe11myu4bkcetslzex6z9c5x95cnnvwdv0h6r4dfbyzoj24lby7y7ynso88dmyv7e8yyqv2fa03n0hgw22icisc8n0poaycfx0cntc1bg0ejgw9a7fv3tgp4u4yz73xsu9mubio3r8nfd2xz4q5r2nfez1609kcdc88r809llvd2b6m4zcbqdlczdan8qxjeja4b4ce5iohzljmij35gg4w95jgel4get68mbsw4wnyq1oy6nwdmnijbfk6my2xlwsgqwsfgx5fx8qv8ps0bi80rft656viwtz7lgf5a92i3z5k60rtm10pruudqzw2tyzpe25i5admveoab3opdcyz3yolw9mjuaxxx9ddkfazfse86bgrw3l0shv365858p078nct02638bnkq855kck1zjqoj8vn03b3cv94om1h7hopg9truzj9626afydpqckf0w53ff82shj6yghahencktxzezfpjujaa0r5tkwln332524oownoligu08w1b07zm6f2lmvj2xp96khohng1zza0bmnbf4znepogoowvxftbd2t0i0c1says0v3ezkkl4k84m80190dc9u890bilixd34kqbsytbjshdb0eploo4t3haxdu3lhcukqn8gac7ouc151vtzhahmk6ge2xjvqldjoob1cdpn9dq1n1yvhh72tahib0glmyvda58wnksq4srfgpwlgmfz3bvayukhdd9nmta7205f1wyqd1fp4e9cxq7fuwu9hd6mciguaekgtsong9fet1ob2afp1nowe6acf7tl5819lnneibmhjg8rw2fbcxeq24fbjgqg4ajblsiy8h99kqk805psfu2yszn8pr4m0z0sqieyg4zjpfi4u0aw6o3ruoun71zw3a0mnpdtlecl38l7iqs3hdbef81fgwm5dm2oyptm552kukfaqvkx4kkgqrc47ngruyvcb3625jormivx9nai8tvfh650ekd9h4wyicdxgcuudndc2r7pq8dff4i90czklyj9sahimcmrtjycix0k699y2uncufnrp0x5kn35pox3vjyed01v7semyrlsd3szdjgnb88lenotvitz112oeraiw4vjin77mdum5uce0idainpsp30xjz3xxmy308dkwe434yxbqpkozbvq2rhfmiulqr4q7o4zqho4o5a89i89bmdiq1giqctd6qg9wxa6jgpzr2ras90msvwz5271o2musgq0uklmg6tuabbw0lq6p67i40u9srrmqkg12ro0vza20qp9ih1nax443bd80a53kbfd9wl6jf1mm00frfj8gbr3fraa64kavttukf49ghvmlhsx164dc6arws6zl4482xj2bhqb6lo0nlp8rrdfrwu3nj15hzs6cukhxiuh3sp2gtthus85avuq3edub36ui84pq3g1nu3u2fn999avoh7mgb4o9afaqjjobd4x3u2z502t2uvaqdg4u52u3v8n8ppbs7jb91mjncnu4qhxdnlvkk9wyq792z3yfq88j9bm2cp5jaercc779mcw6b62yzqwpnssn6tj3y9xx1frujh9eq5tc18z21aa6h8iybj2qjuqm0mkm9paucujj24vh2zher3caouvtxer7p94dyh5hzor2s8hjbmpfcbavabyb7xeamgj9svyo5h0fj3cnjygncx935uiefhn8n89eas1mkh3ptu08c83ldmaz0byuk0mynm4u03jbi9ov99febtpzt8yz0b9eobdbj7ub425cqzj6x0sj02gd3uon8xk3s63l2320b6uwmpsj8dgoco1ttggk0uyysv1luybhqycu73rw9oavjd61opjx5m7mqrnwdbzb1ee9kujovy9h13pab4bncb6k2r5ae91fiv4dd2mupttuupf2ys8g85k9b6shwxuxrlsndebmjn4h2kxohjh6q5wlshp3usbpzjz7nwts85fphxs5fpykrdbro7ttzhu6cun1xse4muk8nz6sm6hg4yjkn9u3idtepc8i7qjt90lguoo0o8ws6ny3dn7kb3m80ej4h0yy5j2a0x13uv8qbwqfjy6yxynma8ymxqtyamlj2mfe16445dn41idg7yn54fiq8mjcepjq4u4dc5xm91xm3gv1t57jvv5qjtti\n\n",
		'error' => NULL,
	)),
))
